src/rpc_client/eth_rpc/mod.rs

```diff
@@ -0,0 +1,851 @@
+//! This module contains definitions for communicating with an Ethereum API using the [JSON RPC](https://ethereum.org/en/developers/docs/apis/json-rpc/)
+//! interface.
+
+use crate::log;
+use crate::memory::next_request_id;
+use crate::rpc_client::checked_amount::CheckedAmountOf;
+use crate::rpc_client::eth_rpc_error::{sanitize_send_raw_transaction_result, Parser};
+use crate::rpc_client::numeric::{BlockNumber, LogIndex, TransactionCount, Wei, WeiPerGas};
+use crate::rpc_client::responses::TransactionReceipt;
+use crate::rpc_client::RpcTransport;
+use candid::candid_method;
+use ethnum;
+use evm_rpc_types::{HttpOutcallError, JsonRpcError, RpcError, RpcService};
+use ic_cdk::api::call::RejectionCode;
+use ic_cdk::api::management_canister::http_request::{
+    CanisterHttpRequestArgument, HttpHeader, HttpMethod, HttpResponse, TransformArgs,
+    TransformContext,
+};
+use ic_cdk_macros::query;
+use ic_ethereum_types::Address;
+use minicbor::{Decode, Encode};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::fmt;
+use std::fmt::{Debug, Display, Formatter, LowerHex, UpperHex};
+
+#[cfg(test)]
+mod tests;
+
+// This constant is our approximation of the expected header size.
+// The HTTP standard doesn't define any limit, and many implementations limit
+// the headers size to 8 KiB. We chose a lower limit because headers observed on most providers
+// fit in the constant defined below, and if there is spike, then the payload size adjustment
+// should take care of that.
+pub const HEADER_SIZE_LIMIT: u64 = 2 * 1024;
+
+// This constant comes from the IC specification:
+// > If provided, the value must not exceed 2MB
+const HTTP_MAX_SIZE: u64 = 2_000_000;
```

Comment:
Note that the maximum size is 2 MiB and not 2 MB (see [here](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/maintain/resource-limits)).

Reply:
Changed [here](https://github.com/internet-computer-protocol/evm-rpc-canister/pull/279/commits/c7634e254f69038a50c4e052f736e8e9a05a29d3)